plugins/modules/writable_snapshots.py

```diff
@@ -0,0 +1,591 @@
+#!/usr/bin/python
+# Copyright: (c) 2024, Dell Technologies
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+""" Ansible module for managing writable snapshots on PowerScale"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: writable_snapshots
+version_added: '3.3.0'
+short_description: Manage writable snapshots on PowerScale
+description:
+- You can perform the following operations.
+- Managing snapshots on PowerScale.
+- Create a writable snapshot.
+- Delete a writable snapshot.
+
+extends_documentation_fragment:
+  - dellemc.powerscale.powerscale
+
+author:
+- Kritika Bhateja(@Kritika-Bhateja-03) <ansible.team.dell.com>
+
+options:
+  writable_snapshots:
+    description:
+    - List of writable snapshots details.
+    required: False
+    type: list
+    elements: dict
+    suboptions:
+      state:
+        description:
+        - The state of the writable snapshot to create or delete.
+        - I(state) is C(present) - To create a writable snapshot.
+        - I(state) is C(absent) - To delete the writable snapshot.
+        default: present
+        choices: ['present', 'absent']
+        type: str
+      dst_path:
+        description:
+        - The /ifs of the writable snapshot.
+        - The destination path should be non-existing path and it's absolute path.
+        required: true
+        type: path
+      src_snap:
+        description:
+        - The source snapshot name or ID.
+        - This option is required I(state) is C(present).
+        required: false
+        type: str
+notes:
+- The I(check_mode) is supported.
```

Comment:
check_mode and diff should be moved to attributes and remove from notes as per the ansible documentation.
https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-block 

@anupamaloke FYI.

Reply:
done